### PR TITLE
First step in adding Tf

### PR DESCRIPTION
**Pull request by dabail10**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
Added Tf to all the subroutine calls and used in place of Tocnfrz.
- [X] Developer(s): 
dabail10 (D. Bailey)
- [X] Suggest PR reviewers from list in the column to the right.
@eclare108213 @apcraig 
- [X] Please copy the PR test results link or provide a summary of testing completed below.
Compared two single cases with and without the change.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:
This is the first step in this change. The next will change answers for the BGC. Note this might also change answers in CICE at least in the history files.

### CODE REVIEW

File: columnphysics/icepack_mechred.F90 [hunk 1]
```diff
@@ -1162,7 +1168,7 @@ subroutine ridge_shift (ntrcr,       dt,              &
 
       real (kind=dbl_kind), dimension(:), intent(inout) :: &
          miso           ! isotope mass added to ocean (kg m-2)
-
+
```
Comment (eclare108213) [inline columnphysics/icepack_mechred.F90]: extra space added here

File: configuration/driver/icedrv_forcing.F90 [hunk 2]
```diff
@@ -1072,10 +1072,11 @@ end subroutine ocn_ISPOL
 
       subroutine finish_ocn_forcing(sst_temp)
 
- ! Compute ocean freezing temperature Tf based on tfrz_option
- ! 'minus1p8'         Tf = -1.8 C (default)
- ! 'linear_salt'      Tf = -depressT * sss
- ! 'mushy'            Tf conforms with mushy layer thermo (ktherm=2)
+! Compute ocean freezing temperature Tf based on tfrz_option
+! 'minus1p8'         Tf = -1.8 C (default)
+!  'constant'         Tf = Tocnfrz from namelist
```
Comment (apcraig) [inline configuration/driver/icedrv_forcing.F90]: extra space to be removed?

File: configuration/driver/icedrv_forcing.F90 [hunk 3]
```diff
@@ -1072,10 +1072,11 @@ end subroutine ocn_ISPOL
 
       subroutine finish_ocn_forcing(sst_temp)
 
- ! Compute ocean freezing temperature Tf based on tfrz_option
- ! 'minus1p8'         Tf = -1.8 C (default)
- ! 'linear_salt'      Tf = -depressT * sss
- ! 'mushy'            Tf conforms with mushy layer thermo (ktherm=2)
+! Compute ocean freezing temperature Tf based on tfrz_option
+! 'minus1p8'         Tf = -1.8 C (default)
```
Comment (eclare108213) [inline configuration/driver/icedrv_forcing.F90]: Should this 'default' comment be changed?

File: doc/source/icepack_index.rst [hunk 4]
```diff
@@ -454,7 +454,7 @@ either Celsius or Kelvin units).
    "tday", "absolute day number", ""
    "Tf", "freezing temperature", "C"
    "Tffresh", "freezing temp of fresh ice", "273.15 K"
-   "tfrz_option", ":math:`\bullet` form of ocean freezing temperature", ""
+   "tfrz_option", ":math:`\bullet` form of ocean freezing temperature", "mushy"
```
Comment (apcraig) [inline doc/source/icepack_index.rst]: Just FYI, it's not clear whether the last column is supposed to be the default or units or something else.  I think leaving this blank is OK, it's also OK to define the default, or probably to say "character string".

File: doc/source/user_guide/ug_case_settings.rst [hunk 5]
```diff
@@ -359,8 +359,9 @@ forcing_nml
    "", "``mm_per_sec``", "(same as MKS units)", ""
    "", "``m_per_sec``", "", ""
    "``restore_ocn``", "logical", "restore sst to data", "``.false.``"
-   "``tfrz_option``", "``linear_salt``", "linear functino of salinity (ktherm=1)", "``mushy``"
+   "``tfrz_option``", "``linear_salt``", "linear function of salinity (ktherm=1)", "``mushy``"
```
Comment (apcraig) [inline doc/source/user_guide/ug_case_settings.rst]: This is where the default should be defined and it is.  Could you reorder the options into alphabetical order.  I know it's a pain, but that means constant would be first.  Thanks!

File: doc/source/icepack_index.rst (hunk 4 above)
Comment (eclare108213) [reply doc/source/icepack_index.rst]: Here in the index, I think the convention is to _not_ put default values from the code unless they are hard-coded and/or are not expected to change (like the freezing temperature of fresh ice), but do put units where possible.  The default values belong in ug_case_settings, as @apcraig notes below.

File: doc/source/user_guide/ug_case_settings.rst [hunk 6]
```diff
@@ -359,7 +359,8 @@ forcing_nml
    "", "``mm_per_sec``", "(same as MKS units)", ""
    "", "``m_per_sec``", "", ""
    "``restore_ocn``", "logical", "restore sst to data", "``.false.``"
-   "``tfrz_option``", "``linear_salt``", "linear functino of salinity (ktherm=1)", "``mushy``"
+   "``tfrz_option``","``constant``", "constant ocean freezing temperature (Tocnfrz)"
+   "", "``linear_salt``", "linear function of salinity (ktherm=1)", "``mushy``", ""
```
Comment (eclare108213) [inline doc/source/user_guide/ug_case_settings.rst]: Is there an extra pair of quotation marks in this line?

Comment (apcraig) [reply doc/source/user_guide/ug_case_settings.rst]: The "mushy" default value needs to be moved to the "constant" line then I think it's OK.

**Review by eclare108213 — CHANGES_REQUESTED** [review]
This is set to be merged into E3SM-Project:main, not what we want to do!  It needs to be a PR into E3SM-Project:cice-consortium/E3SM-icepack-initial-integration

**Comment by dabail10** [comment]
Is anyone else allowed to edit the destination? Or do I need to close and re-open?

**Comment by eclare108213** [comment]
I was able to change it.  But now there are conflicts...

**Comment by apcraig** [comment]
@eclare108213, good catch!

**Comment by dabail10** [comment]
Ok. I will fix this.

**Comment by apcraig** [comment]
I don't understand the conflicts.  Not sure why they're showing up.  @dabail10, please review the PR diffs to make sure you didn't start from a different version of the code.  All the diffs should be recognizable to you.

**Comment by dabail10** [comment]
I had to do a rebase. I guess main is different enough from this branch?

**Comment by dabail10** [comment]
Looks like I have to do this another way.

**Comment by dabail10** [comment]
I tried to merge the branch into my branch tfrz. It was requiring a rebase and then would not let me push it back to tfrz. I was able to create a new branch tfrz2 from the E3SM-Icepack-partial-integration. Then I could merge from tfrz to here. I will close this and reissue from tfrz2.

**Comment by apcraig** [comment]
OK, so you didn't start with [E3SM-Project:cice-consortium/E3SM-icepack-initial-integration](https://github.com/E3SM-Project/Icepack/tree/cice-consortium/E3SM-icepack-initial-integration)?  I'm surprised the PR is working as well as it is then.  For reference, this is what I do to develop the E3SM Icepack.

```
git clone https://github.com/E3SM-Project/Icepack icepack.mybranch
cd icepack.mybranch
git checkout cice-consortium/E3SM-icepack-initial-integration
git branch mybranch
git checkout mybranch
git remote add myfork https://github.com/apcraig/Icepack
# develop and test
git commit ...
git push myfork mybranch
```

I start with the branch I want to work on, create a local branch and develop there.  I then push the branch onto my fork.  Then PR from myfork:mybranch to E3SM-Project:cice-consortium/E3SM-icepack-initial-integration.  I think this is the easiest way to contribute on non-standard forks/branches.

In fact, we maybe should be doing that by default even for the Consortium main.  You need not keep your fork/branch up to date with main and branch off your fork main.  You can just checkout CICE-Consortium:main, create a branch, develop, and then finally push back to your fork.  Then PR back to the Consortium.  The only drawback to that you have to be very careful not to do "git push origin mybranch" at the end or you push directly to CICE-Consortium which we don't ever want.

**Comment by dabail10** [comment]
The tfrz branch had started from main. This one I started from the correct branch.

**Pull request closed without merging**